libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -391,10 +408,40 @@ void IRGenerator::generateImplicitConstructors(ContractDefinition const& _contra
 string IRGenerator::deployCode(ContractDefinition const& _contract)
 {
 	Whiskers t(R"X(
+		<#loadImmutables>
+			let <var> := mload(<memoryOffset>)
+		</loadImmutables>
+
 		codecopy(0, dataoffset("<object>"), datasize("<object>"))
+
+		<#storeImmutables>
+			setimmutable("<immutableName>", <var>)
```

Comment:
I'm still wondering whether this should at some point have a much more general signature, like
```
setimmutable(0 /* memory offset of code to be modified */, "<object>", "<immutableName>", <var>)
```
Then immutable references would be an implicit property of yul objects and the semantics would be clearer even if e.g. there's no unique runtime object or a deploy routine that doesn't write code to zero in memory (although I'd not be sure about details, like if the memory offset should be a literal, etc.)...
But then again we might come up with an even better way in the future and I guess it's fine for now (especially since the assembly wouldn't support the extended signature right away).

Reply:
Referencing the object sounds like a good idea!

Reply:
Maybe we could come up with a more sophisticated `codecopy` function that performs the code copy and the immutable replacement at the same time.

Reply:
I'll create an issue for it.